src/Logic/CoreRules.mli

```diff
@@ -49,6 +49,8 @@ exception NoOpenGoal
 
 val get_proof : proof_state -> proof_tree
 val mk_name : ?name:string -> sec_exp -> string
+val prove_by : rule -> tactic
+                                           
```

Comment:
This breaks the CoreRules abstraction. Rules that require prove_by should be in CoreRules.